modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -1057,9 +1079,8 @@ private CompletableFuture<?> createTableLocally(long causalityToken, int catalog
 
             // Check if the table already has assignments in the vault.
             // So, it means, that it is a recovery process and we should use the vault assignments instead of calculation for the new ones.
-            // TODO: IGNITE-20210 Fix it
-            if (partitionAssignments(vaultManager, tableId, 0) != null) {
-                assignmentsFuture = completedFuture(tableAssignments(vaultManager, tableId, zoneDescriptor.partitions()));
+            if (partitionAssignments(metaStorageMgr, tableId, 0, causalityToken) != null) {
```

Comment:
Why do you use the call for 0 participation only?

Reply:
It's used only for condition to peek whether there are any assignments in meta storage. If yes, we read the assignments themselves below